File: src/March2021/strings/java/NonRepeatingString.java
package March2021.strings.java;

import java.util.Arrays;
import java.util.Scanner;

public class NonRepeatingString {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int numberOfCases = sc.nextInt();
        while (numberOfCases-- > 0) {
            String inputStr = sc.next();

            int result = firstNonRepeatingChar(inputStr);
            if (result >= 0) {
                System.out.println("Non-repeating string is: " + inputStr.charAt(result));
            } else {
                System.out.println("No non-repeating string");
            }
        }
    }

    private static int firstNonRepeatingChar(String inputStr) {
        int[] map = new int[256];
        Arrays.fill(map, 0);
        int x = 0;
        for (int i = 0; i < inputStr.length(); i++) {
            map[inputStr.charAt(i)] = map[inputStr.charAt(i)] + 1;
        }

        for (int i = 0; i < inputStr.length(); i++) {
            if (map[inputStr.charAt(i)] == 1) {
                return i;
            }
        }
        return -1;
    }
}

File: src/March2021/strings/kotlin/NonRepeatingString.kt
package March2021.strings.kotlin

import java.util.*

fun main(args: Array<String>) {
    // Reading input from user
    val inputStr = readLine()
    // changing string into char array
    val inputChar = inputStr!!.toCharArray()

    // making int array of size 256 and populating with -1
    var map = IntArray(256) {-1}
    // for loop with index
    for ((index, value) in inputChar.withIndex()) {
        map[inputChar[index].toInt()] = map[inputChar[index].toInt()] + 1
    }

    // for loop for finishing string
    for (i in inputChar.indices) {
        if (map[inputChar[i].toInt()] == 0) {
            print("The first non-repeating char is: ${inputChar[i]}")
            break;
        }
    }
}